rules/misc.lua

```diff
@@ -395,3 +395,585 @@ rspamd_config.MISSING_FROM = {
+  description = 'Reply-To header could not be parsed',
+  score = 1.0
+}
+
+rspamd_config.HAS_REPLYTO = {
+  callback = function ()
+    -- Set by CHECK_REPLYTO
+  end,
+  description = 'Has Reply-To header',
+  score = 0.0
+}
+
+rspamd_config.REPLYTO_EQ_FROM = {
+  callback = function ()
+    -- Set by CHECK_REPLYTO
+  end,
+  description = 'Reply-To header is identical to From header',
+  score = 0.0
+}
+
+rspamd_config.REPLYTO_ADDR_EQ_FROM = {
+  callback = function ()
+    -- Set by CHECK_REPLYTO
+  end,
+  description = 'Reply-To address is the same as From',
+  score = 0.0
+}
+
+rspamd_config.REPLYTO_DOM_EQ_FROM_DOM = {
+  callback = function ()
+    -- Set by CHECK_REPLYTO
+  end,
+  description = 'Reply-To domain matches the From domain',
+  score = 0.0
+}
+
+rspamd_config.REPLYTO_DOM_NEQ_FROM_DOM = {
+  callback = function ()
+    -- Set by CHECK_REPLYTO
+  end,
+  description = 'Reply-To domain does not match the From domain',
+  score = 0.0
+}
+
+rspamd_config.REPLYTO_DN_EQ_FROM_DN = {
+  callback = function ()
+    -- Set by CHECK_REPLYTO
+  end,
+  description = 'Reply-To display name matches From',
+  score = 0.0
+}
+
+rspamd_config.CHECK_MIME = {
+  callback = function (task)
+    local parts = task:get_parts()
+    if not parts then return false end
+
+    -- Make sure there is a MIME-Version header
+    local mv = task:get_header('MIME-Version')
+    if (not mv) then
+      task:insert_result('MISSING_MIME_VERSION', 1.0)
+    end
+
+    local found_ma = false
+    local found_plain = false
+    local found_html = false
+
+    for _,p in ipairs(parts) do
+      local mtype,subtype = p:get_type()
+      local ctype = mtype:lower() .. '/' .. subtype:lower()
+      if (ctype == 'multipart/alternative') then
+        found_ma = true
+      end
+      if (ctype == 'text/plain') then
+        found_plain = true
+      end
+      if (ctype == 'text/html') then
+        found_html = true
+      end
+    end
+
+    if (found_ma) then
+      if (not found_plain) then
+        task:insert_result('MIME_MA_MISSING_TEXT', 1.0)
+      end
+      if (not found_html) then
+        task:insert_result('MIME_MA_MISSING_HTML', 1.0)
+      end
+    end
+  end
+}
+
+rspamd_config.MISSING_MIME_VERSION = {
+  callback = function ()
+    -- Set by CHECK_MIME
+  end,
+  description = 'MIME-Version header is missing',
+  score = 2.0
+}
+
+rspamd_config.MIME_MA_MISSING_TEXT = {
+  callback = function ()
+    -- Set by CHECK_MIME
+  end,
+  description = 'MIME multipart/alternative missing text/plain part',
+  score = 2.0
+}
+
+rspamd_config.MIME_NA_MISSING_HTML = {
+  callback = function ()
+    -- Set by CHECK_MIME
+  end,
+  description = 'MIME multipart/alternative missing text/html part',
+  score = 2.0
+}
+
+-- Used to be called IS_LIST
+rspamd_config.PREVIOUSLY_DELIVERED = {
+  callback = function(task)
+    if not task:has_recipients(2) then return false end
+    local to = task:get_recipients(2)
+    local rcvds = task:get_header_full('Received')
```

Comment:
You can get this information parsed when using `task:get_received`. Extra searches seems to be unneeded.